### ISSUE TITLE
Change "environment" to "env" in launch.json

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode-cpptools#116]
Moved from https://github.com/Microsoft/vscode/issues/9733:
- VSCode Version: 1.3.1 with C/C++ extension 0.7.1
- OS Version: Linux

Steps to Reproduce: Open launch.json and hover over "environment":
![bildschirmfoto zu 2016-07-26 09-14-41](https://cloud.githubusercontent.com/assets/80071/17129146/ad8f5b5c-5311-11e6-953c-d82ed8d6c60f.png)

I followed the example, but I'm getting `Incorrect type. Expected "array"`. This page https://code.visualstudio.com/docs/editor/debugging says I should use `"env"`, but I guess that's because it's for Node.js debugging and I'm trying to debug a C++ application?

**Comment by pieandcakes** [comment]
@kieferrm the environment variables are name/value pairs within an array. Proper usage should be:
"environment": [{"name": "squid", "value": "clam"}, {"name": "squid1", "value": "clam1"}]

It is expecting an array which is defined by [ ] and then each name/value pair is within { }.

**Comment by jhasse** [comment]
Ah I see thanks :)

What do you think about renaming this to `env` and make it behave just as with node.js debugging? This would make it more intuitive IMHO.

**Comment by pieandcakes** [comment]
@jhasse we can take it into consideration. I feel like `"environment"` is very clear and since it is already there, it doesn't need to be typed :)

I have also updated the help example to include the array brackets.

**Comment by m-mcgowan** [comment]
Yes, so rather than have a consistent way to do the same thing across different languages (since environment isn't part of the language but...well...part of the environment) let's make it different for each language.  And the confusion caused caused by the differences is totally worth not having to type out "environment" because it's already there.

**Comment by pieandcakes** [comment]
@m-mcgowan @jhasse We will consider it for a future release.

**Comment by Chunting** [comment]
Hi,
How can I access the env or args variables in C++ in my program?
For example, I want to access args[1] in my code or "name": "sceneFile", how can I make it happen?
![image](https://user-images.githubusercontent.com/3312409/38343148-49ff7a24-3850-11e8-9d3f-7ecdc6660366.png)

Thanks,
Chunting

**Comment by jhasse** [comment]
Just as you would with normal environment variables (`getenv`) or program arguments (`argv`).

**Comment by nsubtil** [comment]
As long as we're considering changes here, why do we need to express a list of environment variables as an array of associative arrays with a single entry in each? Could we just have an associative array where the key is the environment variable name and the value is the variable value itself? Something like:

```
"environment": {  "_NO_DEBUG_HEAP": "1",  "SEARCH_PATH": "c:/data" }
```

would be a lot easier to deal with. Or even just a plain array of "name=value" strings?

(FWIW, the current syntax is strange enough that I find myself Googling for it every time I need to add an environment variable.)

**Comment by dcermak** [comment]
I'd like to throw in a +1 on this one, as having this consistent across various extensions would be highly beneficial. Especially googling for setting environment variables in launch.json gives you the docs of the C++ extension, which will not work for all other language stacks (and the docs are also not too obviously just applicable for the C++ extension if you just skim them).

**Comment by stefannilsson** [comment]
+1 to @nsubtil suggestion! Spent half an hour trying to figure out why my env vars. was not taking effect.

**Comment by bobbrow** [comment]
@mrx23dot I'm not aware of any recent change to the launch.json schema.

**Comment by bobbrow** [comment]
This issue has been open a long time and there are still no plans to change this, so I will close the issue.
